$NetBSD: patch-meson.build,v 1.2 2021/09/16 13:02:34 schmonz Exp $

* Non-MS Windows case uses libintl for dgettext.
  Fix build under NetBSD.
* Apply proposed upstream commit 390e5f02 to fix non-x86 clang builds.
* Omit unsupported linker arguments on macOS.

--- meson.build.orig	2021-07-27 20:02:27.000000000 +0000
+++ meson.build
@@ -216,7 +216,6 @@ endif
 check_headers = [
   'arpa/inet.h',
   'byteswap.h',
-  'cpuid.h',
   'dlfcn.h',
   'execinfo.h',
   'grp.h',
@@ -276,6 +275,19 @@ if cc.has_header_symbol('pthread.h', 'PT
   cdata.set('HAVE_PTHREAD_PRIO_INHERIT', 1)
 endif
 
+# Headers which are usable
+
+check_usable_headers = [
+  'cpuid.h',
+]
+
+foreach h : check_usable_headers
+  if cc.check_header(h)
+    define = 'HAVE_' + h.underscorify().to_upper()
+    cdata.set(define, 1)
+  endif
+endforeach
+
 # Functions
 
 check_functions = [
@@ -348,7 +360,7 @@ if cc.has_header_symbol('sys/syscall.h',
 endif
 
 if cc.has_function('dgettext')
-  if host_machine.system() != 'windows'
+  if host_machine.system() == 'windows'
     libintl_dep = []
   else
     libintl_dep = cc.find_library('intl')
@@ -404,7 +416,7 @@ cdata.set('MESON_BUILD', 1)
 # so we request the nodelete flag to be enabled.
 # On other systems, we don't really know how to do that, but it's welcome if somebody can tell.
 # Windows doesn't support this flag.
-if host_machine.system() != 'windows'
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
   nodelete_link_args = ['-Wl,-z,nodelete']
 else
   nodelete_link_args = []
